polygon/blocksync/block_downloader.go

```diff
@@ -0,0 +1,132 @@
+package blocksync
+
+import (
+	"context"
+	"errors"
+	"math/big"
+
+	"github.com/ledgerwatch/erigon/core/types"
+)
+
+func NewBlockDownloader(p2pLayer p2pLayer, dbLayer dbLayer, heimdallLayer heimdallLayer) *BlockDownloader {
```

Comment:
```suggestion
func NewDownloader(Sentry sentry, SyncDb db, Heimdall heimdall) *Downloader {
```

A very descriptive name could be "a downloader that uses Heimdall state points to download block headers from Sentry and put them into DB". How can we shorten it? `StatePointDownloader` is somewhat misleading, because it doesn't download state points (it does, but it is not the main purpose). `DownloaderThatUsesStatePoints`? `StatePointAwareDownloader`? 🤔 I suggest to call the class simply `Downloader` for now until we find a better name. `HeaderDownloader` is fine too.

What's the purpose of the "layer" suffix? I suggest to keep argument names without the suffix.

Reply:
(ignore the layer stuff, this was just me putting a dummy interface name so I dont waste time on naming now, this is just a super draft version of the spec we discussed - I'll rename these once we have all the interfaces in place in our new package)

Reply:
addressed